src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const setOfStates = [];
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
Use destructuring for `actiona`
```suggestion
  for (const action of actions) {
    const { type, .... } = action;
    
    switch (type) {
```